packages/adminrouter/extra/src/test-harness/tests/test_service.py

```diff
@@ -553,3 +554,198 @@ def test_if_scheme_is_honoured_in_mesos_scheduler_entry(
+            # slightly higher than: 2 * (backend_request_timeout * 0.5)
+            assert t_spent > backend_request_timeout * 0.5
+
+    def test_if_marathon_borkerage_prevents_resolving_via_mesos_state_summary(
+            self, mocker, nginx_class, valid_user_header):
+        # Bork Marathon Mock, DO NOT touch Mesos Mock:
+        mocker.send_command(
+            endpoint_id='http://127.0.0.1:8080',
+            func_name='always_bork',
+            aux_data=True)
+
+        # Make period cache refreshes so rare that they do not get into
+        # picture:
+        ar = nginx_class(cache_first_poll_delay=1200,
+                         cache_poll_period=1200,
+                         cache_expiration=1200,
+                         cache_max_age_soft_limit=1200,
+                         cache_max_age_hard_limit=1800,
+                         )
+        url = ar.make_url_from_path("/service/scheduler-alwaysthere/")
+
+        with GuardedSubprocess(ar):
+            resp = requests.get(
+                url,
+                allow_redirects=False,
+                headers=valid_user_header)
+
+        assert resp.status_code == 503
+        assert '503 Service Unavailable: invalid Marathon svcapps cache' in resp.text
+
+    def test_if_mesos_borkerage_prevents_resolving_via_mesosdns(
+            self, mocker, nginx_class, valid_user_header):
+        # Bork Mesos Mock, Make Marathon mock respond with no apps, so that AR
+        # tries to resolve via Mesos /state-summary
+        mocker.send_command(
+            endpoint_id='http://127.0.0.1:8080',
+            func_name='set_apps_response',
+            aux_data={"apps": []})
+        mocker.send_command(
+            endpoint_id='http://127.0.0.2:5050',
+            func_name='always_bork',
+            aux_data=True)
+
+        # Make period cache refreshes so rare that they do not get into
+        # picture:
+        ar = nginx_class(cache_first_poll_delay=1200,
+                         cache_poll_period=1200,
+                         cache_expiration=1200,
+                         cache_max_age_soft_limit=1200,
+                         cache_max_age_hard_limit=1800,
+                         )
+        url = ar.make_url_from_path("/service/scheduler-alwaysthere/")
+
+        with GuardedSubprocess(ar):
+            resp = requests.get(
+                url,
+                allow_redirects=False,
+                headers=valid_user_header)
+
+        assert resp.status_code == 503
+        assert '503 Service Unavailable: invalid Mesos state cache' in resp.text
+
+    def test_if_mesos_borkerage_does_not_prevent_resolving_via_marathon(
+            self, mocker, nginx_class, valid_user_header):
+        # Bork Mesos Mock, Make MesosDNS mock respond with no apps, so that AR
+        # is able to resolve only via Marathon/we are certain that it resolved
+        # via Marathon.
+        mocker.send_command(
```

Comment:
I was thinking about it at the very begining. The problem is that we have ~38 mock endpoints, giving name to each one of them would be even more confusing. This is why we have this idea of `mocker` which is basically a control plane on top of all these mock endpoints (mock HTTP servers to be precise).

Also, we need to be very carefull to not to overdoo and not to introduce too many terms. People from outside are familiar with the Mesos socket, but not necessarily with `mesos_dns_endpoint` term. 

Even more - I think I have three different Marathons, all of them used for different tests. What you are suggesting would require being extra creative with naming or using `marathon_endpoint_1`, marathon_endpoint_2`, `marathon_endpoint_default`, etc...